### PR TITLE
[KOGITO-3708] - Decreasing the amount of reconciliation loops for Kog…

### PR DESCRIPTION
**Pull request by ricardozanini**
…itoInfra

See: https://issues.redhat.com/browse/KOGITO-3708

This is what looks like a non-OK infra resource:

```
oc get kogitoinfra
NAME                      RESOURCE NAME   KIND         API VERSION         STATUS   REASON
kogito-infinispan-infra                   Infinispan   infinispan.org/v1   False    ResourceAPINotFound
```

Here's the interval between reconciliations:

```
2020-10-28T17:05:36.365-0300	INFO	kogitoinfra_controller	Successfully Update Kogito Infra status
...
2020-10-28T17:06:06.338-0300	INFO	kogitoinfra_controller	Reconciling KogitoInfra for kogito-infinispan-infra in kogito-3708
```

30 seconds :)

Signed-off-by: Ricardo Zanini <zanini@redhat.com>

Many thanks for submiting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors' guide](https://github.com/kiegroup/kogito-cloud-operator/blob/master/README.md#contributing-to-the-kogito-operator)
- [x] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [x] Your feature/bug fix has a unit test that verifies it
- [x] You've tested the new feature/bug fix in an actual OpenShift cluster

### CODE REVIEW

**Comment by r00ta** [comment]
Hi @ricardozanini , the BDD tests are failing on every PRs (on a data-index test). So I think this is not related to my changes. Can we move forward with this?

**Comment by sutaakar** [comment]
The Data index tag was changed yesterday, images were built few hours ago. Retriggering the tests:
/jenkins test

File: examples/kogitoinfra-infinispan.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#Infinispan operator should be pre-installed in namespace
```
Comment (radtriste) [inline examples/kogitoinfra-infinispan.yaml:1]: If you introduce the infinispan kogitoinfra as standalone yaml file, can you do the same for kafka and keycloak ?

**Review by radtriste — APPROVED** [review]
small suggestion. otherwise looks good

**Review by sutaakar — APPROVED** [review]
Looks good to me, thanks.
Tested and works as expected.

**Comment by ricardozanini** [comment]
@spolti @vaibhavjainwiz could you please take a look? I'll rebase the RELEASE_NOTES doc but this should not affect this PR anyhow.

File: RELEASE_NOTES.md [hunk 2]
```diff
@@ -1,12 +1,14 @@
 ## Enhancements
 
 - [KOGITO-2562](https://issues.redhat.com/browse/KOGITO-2562) - Knative Eventing resources created for Kogito Applications that uses [Knative Addon](https://github.com/kiegroup/kogito-examples/tree/stable/process-knative-quickstart-quarkus).
-- [KOGITO-3094](Provide data-index-infinispan/mongodb as independent images) - Provide data-index-infinispan/mongodb as independent images
+- [KOGITO-3094](https://issues.redhat.com/browse/KOGITO-3094) - Provide data-index-infinispan/mongodb as independent images
```
Comment (spolti) [inline RELEASE_NOTES.md:4]: ooops, thanks for fixing it :D

**Review by spolti — APPROVED** [review]

**Comment by ricardozanini** [comment]
Was working this morning..

/jenkins test

**Comment by sutaakar** [comment]
This time the test failed on Strimzi operator installation:
```
2020-10-29 16:03:10  | 2020-10-29 16:03:10  |        | strimzi-cluster-operator.v0.20.0.164282089666bcd5      | -     | -                                       | Normal  | RequirementsNotMet   | -       | -           | -         | one or more requirements couldn't be found                                                                                                        | 
```
Not sure why, locally it seems that Strimzi is installed successfully.
/jenkins test

**Comment by ricardozanini** [comment]
Data Index test still hangs in the Strimzi operator.

**Comment by ricardozanini** [comment]
This lint error is super weird, locally I can't see anything, my tree is clean. Maybe because we are removing and creating the file everytime? This shouldn't be a problem since the file is the same!

**Comment by radtriste** [comment]
\jenkins test

**Comment by radtriste** [comment]
Having the same problem there: https://github.com/kiegroup/kogito-cloud-operator/pull/639

**Comment by ricardozanini** [comment]
@sutaakar as per our conversation, I'm merging this one.

**Pull request merged**